Sources/Aardvark/Bug Reporting/LogStoreAttachmentGenerator.swift

```diff
@@ -0,0 +1,142 @@
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+@objc(ARKLogStoreAttachmentGenerator)
+public final class LogStoreAttachmentGenerator: NSObject {
+
+    // MARK: - Public Types
+
+    public struct Attachments {
+
+        public var logMessagesAttachment: ARKBugReportAttachment
+
+        public var latestScreenshotAttachment: ARKBugReportAttachment?
+
+    }
+
+    // MARK: - Public Static Methods
+
+    /// Generates bug report attachments representing the data in the log store.
+    ///
+    /// - parameter logStore: The log store from which to read the messages.
+    /// - parameter messageFormatter: The formatter used to format messages in the logs attachment.
+    /// - parameter includeLatestScreenshot: Whether an attachment should be generated for the last screenshot in the
+    /// log store, if one exists.
+    /// - parameter completionQueue: The queue on which the completion should be called.
+    /// - parameter completion: The completion to be called once the attachments have been generated.
+    public static func attachments(
+        for logStore: ARKLogStore,
+        messageFormatter: ARKLogFormatter = ARKDefaultLogFormatter(),
+        includeLatestScreenshot: Bool,
+        completionQueue: DispatchQueue,
+        completion: @escaping (Attachments) -> Void
+    ) {
+        logStore.retrieveAllLogMessages { logMessages in
+            let screenshotAttachment: ARKBugReportAttachment?
+            if includeLatestScreenshot {
+                screenshotAttachment = attachmentForLatestScreenshot(in: logMessages, logStoreName: logStore.name)
+            } else {
+                screenshotAttachment = nil
+            }
+
+            let logsAttachment = attachment(
+                for: logMessages,
+                using: messageFormatter,
+                logStoreName: logStore.name
+            )
+
+            completionQueue.async {
+                completion(
+                    .init(
+                        logMessagesAttachment: logsAttachment,
+                        latestScreenshotAttachment: screenshotAttachment
+                    )
+                )
+            }
+        }
+    }
+
+    /// Generates an attachment containing the latest screenshot in the provided log messages.
+    ///
+    /// - parameter logMessages: The log messages through which to search for the screenshot.
+    /// - parameter logStoreName: The name of the log store from which the logs were collected.
+    @objc(attachmentForLatestScreenshotInLogMessages:logStoreName:)
+    public static func attachmentForLatestScreenshot(
+        in logMessages: [ARKLogMessage],
+        logStoreName: String?
+    ) -> ARKBugReportAttachment? {
```

Comment:
It's the reverse: the previous method is a convenience wrapper around this method and the one below. The previous method is async because it needs to asynchronously load the messages from the log store before calling the separate methods to generate each attachment. Each of the generation methods then taking an array of log messages, which it can then process synchronously.

Reply:
ahhhhhhhhhhhh then the answer was "my understanding is wrong" 😅